tests/evaluate/test_evaluate.py

```diff
@@ -120,14 +121,38 @@ def test_evaluate_call_bad():
     assert score == 0.0
 
 
+@pytest.mark.parametrize(
+    "program_with_example",
+    [
+        (Predict("question -> answer"), new_example("What is 1+1?", "2")),
+        (
+            # Create a program that extracts entities from text and returns them as a list,
+            # rather than returning a Predictor() wrapper. This is done intentionally to test
+            # the case where the program does not output a dictionary-like object because
+            # Evaluate() has failed for this case in the past
+            lambda text: TypedPredictor("text: str -> entities: List[str]")(text=text).entities,
+            dspy.Example(text="United States", entities=["United States"]).with_inputs("text"),
```

Comment:
@okhat This appears to be a valid setup for DSPy evaluation - i.e. all other parts of evaluation work with such a `lambda` except for displaying table output. Let me know if there's a strong reason to disallow this and require all DSPy modules / programs to return `Prediction()` objects